src/Socket.c

```diff
@@ -1356,6 +1361,58 @@ int Socket_new(const char* addr, size_t addr_len, int port, SOCKET* sock)
 	return rc;
 }
 
+#if defined(UNIXSOCK)
```

Comment:
No, this `int Socket_unix_new(const char* addr, size_t addr_len, SOCKET* sock)` should be available for all OS ; but in case of WIN32, it should return a PAHO error or a SOCKET_ERROR with a log that Unix socket is not available.

Doing so, you'll remove many `#if defined(UNIXSOCK).

Reply:
I agree with you that all the conditional compilation in these files makes it quite difficult to follow the flow through this code. But I think, at this point, it is probably be better to be consistent with existing conventions than start doing things differently for a small portion of the code. And, realistically, the inline code for this feature is trivial compared to the rest of what's there.